AK/URLParser.cpp

```diff
@@ -400,293 +517,576 @@ URL URLParser::parse(StringView raw_input, Optional<URL> const& base_url, Option
                 continue;
             }
             break;
+        // -> special authority slashes state, https://url.spec.whatwg.org/#special-authority-slashes-state
         case State::SpecialAuthoritySlashes:
+            // 1. If c is U+002F (/) and remaining starts with U+002F (/), then set state to special authority ignore slashes state and increase pointer by 1.
             if (code_point == '/' && get_remaining().starts_with("/"sv)) {
                 state = State::SpecialAuthorityIgnoreSlashes;
                 ++iterator;
-            } else {
+            }
+            // 2. Otherwise, special-scheme-missing-following-solidus validation error, set state to special authority ignore slashes state and decrease pointer by 1.
+            else {
                 report_validation_error();
                 state = State::SpecialAuthorityIgnoreSlashes;
                 continue;
             }
             break;
+        // -> special authority ignore slashes state, https://url.spec.whatwg.org/#special-authority-ignore-slashes-state
         case State::SpecialAuthorityIgnoreSlashes:
+            // 1. If c is neither U+002F (/) nor U+005C (\), then set state to authority state and decrease pointer by 1.
             if (code_point != '/' && code_point != '\\') {
                 state = State::Authority;
                 continue;
-            } else {
+            }
+            // 2. Otherwise, special-scheme-missing-following-solidus validation error.
+            else {
                 report_validation_error();
             }
             break;
+        // -> authority state, https://url.spec.whatwg.org/#authority-state
         case State::Authority:
+            // 1. If c is U+0040 (@), then:
             if (code_point == '@') {
+                // 1. Invalid-credentials validation error.
                 report_validation_error();
+
+                // 2. If atSignSeen is true, then prepend "%40" to buffer.
                 if (at_sign_seen) {
                     auto content = buffer.to_deprecated_string();
                     buffer.clear();
                     buffer.append("%40"sv);
                     buffer.append(content);
                 }
+
+                // 3. Set atSignSeen to true.
                 at_sign_seen = true;
+
                 StringBuilder builder;
+
+                // 4. For each codePoint in buffer:
                 for (auto c : Utf8View(builder.string_view())) {
```

Comment:
wait hang on a second, just going through the change set, this seems like an obvious typo in the implementation, will investigate